lib/storage/tree.rb

```diff
@@ -0,0 +1,54 @@
+require "tty-tree"
+
+module Storage
+  class Tree
+    attr_reader :name, :children
+  
+    def initialize(name, children)
+      @name = name
+      @children = children
+    end
+    
+    # Convert to a format tty-tree can render properly
+    def toTTY
+      tty_children = []
+      @children.each do |child|
+        if child.class == String
+          tty_children << child
+        else
+          tty_children << child.toTTY
+        end
+      end
+      return { @name => tty_children }
+    end
+    
+    def show
+      TTY::Tree[self.toTTY].render()
+    end
+    
+    # Given the name of a directory in this tree, return a tree rooted at that directory
+    def subtree(dir)
+      @children.each do |tree|
+        if tree.name == dir
+          return tree
+        end
+      end
+      raise "The directory '#{dir}' could not be found"
+    end
+    
+    # Takes a path to a file and returns whether it exists in this tree
+    def fileExists?(path)
+      names = path.split("/").reject { |f| f.empty? }
+      self.exists?(names[0..-2], names.last)
+    end
+    
+    # Don't call this externally, used alongside fileExists?

```

Comment:
"Don't call this externally" looks like a job for the `private` keyword. See: [how to use private](https://www.rubyguides.com/2018/10/method-visibility/); [style guide](https://github.com/rubocop/ruby-style-guide#visibility)

Reply:
It seems that `protected` is what I'm looking for. `exists?` _is_ actually called by other objects, I just don't want any confusion between that and `file_exists?` which is intended to be used to check for a given filepath.